docs/designs/concepts.md

```diff
@@ -0,0 +1,164 @@
+# Concepts Design Doc
+
+This document describes core concepts used in ElasticDL, and discusses their representations.
+
+## Core Concepts
+
+Fundamentally, machine learning is using algorithms to extract information from raw data and represent it in some type of model.
```

Comment:
This sentence is a little confusing

Reply:
Referenced from Wikipedia
`Machine learning algorithms build a mathematical model based on sample data, known as "training data", in order to make predictions or decisions without being explicitly programmed to perform the task.`

Reply:
Remove it :)